### PR TITLE
Removed unexpected maneuvers at sharp bends

### PR DESCRIPTION
**Pull request by dgearhart**
# Issue

Removed unexpected maneuvers at sharp bends and undesirable intersecting edges

### 1. Collapse maneuver at path
image

``` diff
< BEFORE
< 1: Drive southeast on 船原西浦高原線/127.
< 2: Turn right to stay on 船原西浦高原線/127.
< 3: You have arrived at 船原西浦高原線.
> AFTER
> 1: Drive southeast on 船原西浦高原線/127.
> 2: You have arrived at 船原西浦高原線.
```
### 2. Collapse maneuver at footway
image

``` diff
< BEFORE
< 1: Drive southeast on 富士見パークウェイ/Fujimi Parkway.
< 2: Turn left to stay on 富士見パークウェイ/Fujimi Parkway.
< 3: You have arrived at 富士見パークウェイ.
> AFTER
> 1: Drive southeast on 富士見パークウェイ/Fujimi Parkway.
> 2: You have arrived at 富士見パークウェイ.
```

### 3. Collapse maneuver at track
image

``` diff
< BEFORE
< 1: Drive west on 下田石廊松崎線/16.
< 2: Turn left to stay on 下田石廊松崎線/16.
< 3: You have arrived at 下田石廊松崎線.
> AFTER
> 1: Drive west on 下田石廊松崎線/16.
> 2: You have arrived at 下田石廊松崎線.
```

---

## Tasklist

 - [x] Add tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [x] Update the [changelog](CHANGELOG.md)

### CODE REVIEW

File: src/odin/maneuversbuilder.cc [hunk 1]
```diff
@@ -2045,13 +2045,22 @@ bool ManeuversBuilder::CanManeuverIncludePrevEdge(Maneuver& maneuver, int node_i
   // Intersecting forward edge
   if (IsIntersectingForwardEdge(node_index, prev_edge.get(), curr_edge.get())) {
     maneuver.set_intersecting_forward_edge(true);
+    LOG_TRACE("IntersectingForwardEdge");
```
Comment (dgearhart) [inline src/odin/maneuversbuilder.cc:2048]: left trace in on purpose

File: src/odin/maneuversbuilder.cc [hunk 2]
```diff
@@ -2045,13 +2045,22 @@ bool ManeuversBuilder::CanManeuverIncludePrevEdge(Maneuver& maneuver, int node_i
...
+      StreetNamesFactory::Create(trip_path_->GetCountryCode(node_index), prev_edge->GetNameList());
+  std::unique_ptr<StreetNames> common_base_names =
+      prev_edge_names->FindCommonBaseNames(maneuver.street_names());
+
   /////////////////////////////////////////////////////////////////////////////
   // Process 'T' intersection
-  if (IsTee(node_index, prev_edge.get(), curr_edge.get())) {
+  if (IsTee(node_index, prev_edge.get(), curr_edge.get(), !common_base_names->empty())) {
     maneuver.set_tee(true);
+    LOG_TRACE("T intersection");
```
Comment (dgearhart) [inline src/odin/maneuversbuilder.cc:2063]: left trace in on purpose

File: src/odin/maneuversbuilder.cc [hunk 3]
```diff
@@ -2062,17 +2071,13 @@ bool ManeuversBuilder::CanManeuverIncludePrevEdge(Maneuver& maneuver, int node_i
                                     prev_edge->end_heading(), prev_edge->travel_mode())) &&
       !node->HasForwardTraversableIntersectingEdge(prev_edge->end_heading(),
                                                    prev_edge->travel_mode()) &&
-      node->HasTraversableIntersectingEdge(prev_edge->travel_mode())) {
+      node->HasTraversableExcludeUseXEdge(prev_edge->travel_mode(), TripLeg_Use_kTrackUse)) {
+    LOG_TRACE("Non-forward transition with intersecting traversable edge");
```
Comment (dgearhart) [inline src/odin/maneuversbuilder.cc:2075]: left trace in on purpose

**Review by gknisely — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -30,6 +30,7 @@
    * FIXED: Use kServiceRoad edges while searching for ferry connection [#2933](https://github.com/valhalla/valhalla/pull/2933)
    * FIXED: Enhanced logic for IsTurnChannelManeuverCombinable [#2952](https://github.com/valhalla/valhalla/pull/2952)
    * FIXED: Restore compatibility with gcc 6.3.0, libprotobuf 3.0.0, boost v1.62.0 [#2953](https://github.com/valhalla/valhalla/pull/2953)
+   * FIXED: Removed unexpected maneuvers at sharp bends [#2968](https://github.com/valhalla/valhalla/pull/2968)
```
Comment (danpaz) [inline CHANGELOG.md:45]: This PR has less to do with sharpness than with excluding track use roads for maneuvers, right?
```suggestion
   * FIXED: Removed unexpected maneuvers at track roads [#2968](https://github.com/valhalla/valhalla/pull/2968)
```

Comment (dgearhart) [reply CHANGELOG.md:45]: yes and no - more than just tracks - other different travel modes too
All of the issues were at sharp bends

Comment (danpaz) [reply CHANGELOG.md:45]: I see the call to `is_forward` now, I'd missed that. 👍

**Pull request merged**